relay-event-normalization/src/normalize/span/description/mod.rs

```diff
@@ -193,51 +201,99 @@ fn scrub_file(description: &str) -> Option<String> {
     }
 }
 
-/// Normalize the domain and port of a URL.
+/// Scrub a `Url::Host` object. For domain names, run the string through a scrubber. For IP addresses, return the IP address as-is.
+///
+/// # Arguments
+///
+/// * `host` - `Url::Host`
+///
+/// # Returns
+///
+/// A host `String`, or `None` if scrubbing fails.
+pub fn scrub_host(host: Host<&str>) -> Cow<'static, str> {
+    match host {
+        Host::Ipv4(ip) => scrub_ipv4(ip),
+        Host::Ipv6(ip) => scrub_ipv6(ip),
+        Host::Domain(domain) => Cow::Owned(scrub_domain_name(String::from(domain))),
+    }
+}
+
+/// Scrub an IPV4 address. Allow well-known IPs like loopback, and fully scrub out all other IPs.
+///
+/// # Arguments
+///
+/// * `ip` - IP to be scrubbed
+///
+/// # Returns
+///
+/// Scrubbed string
+pub fn scrub_ipv4(ip: Ipv4Addr) -> Cow<'static, str> {
+    if IPV4_ALLOW_LIST.contains(&ip) {
+        return Cow::Owned(ip.to_string());
```

Comment:
It's possible to never allocate here and let the function also return `&'static str` without the `Cow` by storing the ip addresses as `&[(Ipv4Addr, &'static str)]`, e.g. `&[(Ipv4Addr::LOCALHOST, "127.0.0.1")]`.

Something like this:
```rs
IPV4_ALLOW_LIST.into_iter().find_map(|(s, repr)| (s == ip).then_some(repr)).unwrap_or("*.*.*.*")
```

Reply:
I actually considered this, but thought it might get a bit convoluted. Though since both allowlists only contain a single value, we could probably also hard code here, i.e. `if ip == Ipv4Addr::LOCALHOST { return "127.0.0.1" }`.

Reply:
Or just have a big match instead of the static list?